src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -0,0 +1,192 @@
+import React, { useState, useRef } from 'react';
+import styled from 'styled-components';
+import { Button } from '../Button/Button';
+import { mockedAuthorsList, mockedCoursesList } from '../../utils/constants';
+import { timeConverter, createDate, makeid } from '../../utils/functions';
+
+const StyledTop = styled.div`
+	margin: 10px 10px;
+	display: flex;
+	justify-content: space-between;
+`;
+
+const DescriptionInput = styled.input`
+	min-width: 90%;
+	min-height: 100px;
+	margin: 10px 10px;
+`;
+
+const StyledAuthors = styled.div`
+	margin-top: 20px;
+	display: flex;
+	justify-content: space-between;
+`;
+
+const StyledAuthorsList = styled.div`
+	display: flex;
+	margin: 0 auto;
+`;
+
+const StyledLeftColumn = styled.div`
+	width: 40%;
+`;
+
+const StyledRightColumn = styled.div`
+	width: 40%;
+`;
+
+const InputAuthor = styled.input`
+	width: 100%;
+`;
+
+const CreateCourse = ({ close }) => {
+	const authorInput = useRef(null);
+	const [switcher, setSwitcher] = useState(false);
+	const [duration, setDuration] = useState('');
+	const [courseAuthor, setCourseAuthor] = useState([]);
+	const [authorList, setAuthorList] = useState(mockedAuthorsList);
+	const [title, setTitle] = useState('');
+	const [descr, setDescr] = useState('');
+
+	const handleTitle = (event) => {
+		setTitle(event.target.value);
+	};
+
+	const handleDescr = (event) => {
+		setDescr(event.target.value);
+	};
+
+	const newCourse = {
+		id: makeid(20),
+		title,
+		description: descr,
+		creationDate: createDate(),
+		duration,
+		authors: courseAuthor,
+	};
+
+	const addAuthor = (event) => {
+		event.preventDefault();
+		mockedAuthorsList.push({
```

Comment:
please, avoid array mutations in the world of React. See my comment below for details